### PR TITLE
[controller][server][da-vinci-client] Mark a helix replica to error state during ingestion error

### PR DESCRIPTION
**Pull request by majisourav99**
## Mark a helix replica to error state during ingestion error

If ingestion task encounters any error, it just marks the CV to error state which never remediates. This PR makes user of a new [API](https://github.com/apache/helix/pull/2792) helix added to annotate a replica to ERROR state. This will later be picked by the error replica reset task `ErrorPartitionResetTask`  which will attempt to recover those replicas. 
Also added system stores to the controller error replica reset task.

## How was this PR tested?

GHCI
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java [hunk 1]
```diff
@@ -4150,6 +4166,15 @@ public void reportError(String message, int userPartition, Exception e) {
       pcsList.add(partitionConsumptionStateMap.get(userPartition));
     }
     ingestionNotificationDispatcher.reportError(pcsList, message, e);
+    // Set the replica state to ERROR so that the controller can attempt to reset the partition.
+    if (!isDaVinciClient) {
+      zkHelixAdmin.get()
+          .setPartitionsToError(
```
Comment (gaojieliu) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java:4172]: This is to set `EV` state as `ERROR`, right?

Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java:4172]: I saw you mentioned that this logic would only mark `EV` state as `ERROR` when `CV` is in `ERROR` state, so how can we achieve that by the current logic?
Only the following logic today decides whether to propagate `ERROR` to `CV/EV` IIUC.

**Review by gaojieliu — COMMENTED** [review]
Left a few comments.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/HelixParticipationService.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public class HelixParticipationService extends AbstractVeniceService
   private final String clusterName;
   private final String participantName;
   private final String zkAddress;
-  private final StoreIngestionService ingestionService;
+  private StoreIngestionService ingestionService = null;
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/HelixParticipationService.java:82]: I needed it to make integ test work, but now looks like I do not need it.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java [hunk 3]
```diff
@@ -525,6 +533,14 @@ public StoreIngestionTask(
     }
     this.batchReportIncPushStatusEnabled = !isDaVinciClient && serverConfig.getBatchReportEOIPEnabled();
     this.parallelProcessingThreadPool = builder.getAAWCWorkLoadProcessingThreadPool();
+    this.zkHelixAdmin = Lazy.of(() -> new ZKHelixAdmin(zkAddress));
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java:536]: Good point! let me make it a singleton object for all `SIT` to share

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java [hunk 4]
```diff
@@ -114,7 +115,9 @@ public ActiveActiveStoreIngestionTask(
       int errorPartitionId,
       boolean isIsolatedIngestion,
       Optional<ObjectCacheBackend> cacheBackend,
-      DaVinciRecordTransformerFunctionalInterface recordTransformerFunction) {
+      DaVinciRecordTransformerFunctionalInterface recordTransformerFunction,
+      Lazy<ZKHelixAdmin> zkHelixAdmin,
+      int port) {
```
Comment (gaojieliu) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java:120]: `VeniceStoreVersionConfig` can be used to extract `listenerPort` if necessary, so we don't need to explicitly pass this param.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java (hunk 1 above)
Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java:4183]: Shall we add a config to control this behavior?

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java (hunk 4 above)
Comment (majisourav99) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java]: Sure

**Review by gaojieliu — DISMISSED** [review]

**Review by gaojieliu — APPROVED** [review]
Thanks!

**Pull request merged**